src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -472,20 +472,19 @@ private async Task OnPartialValidationSucceededAsync(ChainedHeader chainedHeader
                     }
                 }
 
-                // Partial validation should continue if:
-                //  1: There are more blocks to validate.
-                //  2: A full validation did not fail (if full validation was needed).
-                bool nextValidationNeeded = chainedHeaderBlocksToValidate != null;
-                bool fullValidationFailed = (connectBlocksResult != null) && !connectBlocksResult.Succeeded;
-                if (nextValidationNeeded && !fullValidationFailed)
+                // If more blocks are available continue validation.
+                if (chainedHeaderBlocksToValidate != null)
```

Comment:
I always prefer to rather return here than having nested ifs. So how about:

```
if (chainedHeaderBlocksToValidate = null)
{    
    this.logger.LogTrace("(-)[NO_FURTHER_BLOCKS]");
    return;
}
```

Reply:
hmmm to me this is more readable because we act on the fact its not null, but I am not sure if this entire change is more readable then before, its a suggestion.

Reply:
Its definitely an improvement @dangershony ;)